packages/client/src/components/TestContextChild/index.tsx

```diff
@@ -0,0 +1,7 @@
+import React from "react";
+import { useTest } from "../../contexts";
+
+export default function TestContextChild() {
+  const { connectionMessage } = useTest();
```

Comment:
Every render that the `TestContext` changes (and **only** the ones where it changes), `connectionMessage` will be updated. What's nice about this is we don't have to worry about passing down a ton of props through huge trees, and we can keep global state high up in the tree, while accessing it at very low levels.

Sometimes it's still nice to take this type of context data, map it out into some props, and pass those to various children. It depends on your use case. Context just gives us more flexibility. Bit more on this below.